docs/custom-flows/forgot-password.mdx

```diff
@@ -1,36 +1,52 @@
 ---
-title: Forgot password
-description: Create a custom forgot password flow for your users.
+title: Create a custom Forgot Password flow using Clerk's API
+description: Create a custom forgot password flow for your users using the lower level methods provided by the ClerkJS SDK.
 ---
 
-# Forgot password
+# Create a custom Forgot Password flow using Clerk's API
 
-Clerk's [prebuilt components](/docs/components/overview) have options that you can configure to help your users who have forgotten a password.
+Clerk's [prebuilt components](/docs/components/overview) provide a **Forgot Password** flow for your users out of the box. However, if you want more control over the user experience, you can create a custom **Forgot Password** flow using the lower-level methods provided by the [ClerkJS SDK](/docs/references/javascript/overview).
 
-If these options will not work for you, or if you want more control of your user's experience, it's possible to create a completely custom **Forgot Password** flow using lower-level methods.
+In the following example, the user is asked to provide their email address. After submitting their email, the user is asked to provide a new password and the password reset code that was sent to their email. The user is then signed in with their new password.
 
-## Example
+Note that this example does not handle 2FA (two-factor authentication). If you have 2FA enabled, you will need to handle that separately.
 
-<CodeBlockTabs type="framework" options={["React"]}>
-```jsx
+For the sake of this guide, this example is written for Next.js App Router but it is supported by any React meta framework, such as Remix or Gatsby.
+
+{/* TODO: Add JavaScript example. */}
+
+<CodeBlockTabs type="framework" options={["Next.js"]}>
+```tsx filename="app/forgot-password.tsx"
+"use client"
 import React, { SyntheticEvent, useState } from 'react';
-import { useSignIn } from '@clerk/nextjs';
+import { useAuth, useSignIn } from '@clerk/nextjs';
 import type { NextPage } from 'next';
+import { redirect } from 'next/navigation';
 
-const SignInPage: NextPage = () => {
+const ForgotPasswordPage: NextPage = () => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [code, setCode] = useState('');
   const [successfulCreation, setSuccessfulCreation] = useState(false);
   const [complete, setComplete] = useState(false);
   const [secondFactor, setSecondFactor] = useState(false);
+  const [error, setError] = useState('');
 
+  const { isSignedIn } = useAuth();
   const { isLoaded, signIn, setActive } = useSignIn();
 
   if (!isLoaded) {
     return null;
   }
 
+  // If the user is already signed in, 
+  // and they haven't just completed a password reset, 
+  // redirect them to the home page
+  if (isSignedIn && !complete) {
```

Comment:
The second part (`!complete`) would always return `true` on initial page load since that `useState()` is initialized as false. 

I would probably suggest just removing this, or just doing `if (isSignedIn ) {}`.